src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java

```
@@ -0,0 +1,113 @@
+   *
+   * @return the resource that was added
+   */
+  public static GitRepo addReferenced(AddGitRepoParams addGitRepoParams) {
+    validateEnvironmentVariableName(addGitRepoParams.resourceFields.name);
+
+    // call WSM to add the reference. use the pet SA credentials instead of the end user's
+    // credentials, because they include the cloud-platform scope. WSM needs the cloud-platform
+    // scope to perform its access check before adding the reference. note that this means a user
+    // cannot add a reference unless their pet SA has access to it.
+    GitRepoResource addedResource =
+        WorkspaceManagerService.fromContextForPetSa()
+            .createReferencedGitRepo(Context.requireWorkspace().getId(), addGitRepoParams);
+    logger.info("Created Git repo reference: {}", addedResource);
+    // convert the WSM object to a CLI object
+    Context.requireWorkspace().listResourcesAndSync();
+    return new GitRepo(addedResource);
+  }
+
+  /** Update a Git repo referenced resource in the workspace. */
+  public void updateReferenced(UpdateReferencedGitRepoParams updateParams) {
+    if (updateParams.resourceFields.name != null) {
+      validateEnvironmentVariableName(updateParams.resourceFields.name);
+    }
+    if (updateParams.gitRepoUrl != null) {
+      this.gitRepoUrl = updateParams.gitRepoUrl;
+    }
+    WorkspaceManagerService.fromContext()
+        .updateReferencedGitRepo(Context.requireWorkspace().getId(), id, updateParams);
+    super.updatePropertiesAndSync(updateParams.resourceFields);
+  }
+
+  /** Delete a Git repo referenced resource in the workspace. */
+  protected void deleteReferenced() {
+    // call WSM to delete the reference
+    WorkspaceManagerService.fromContext()
+        .deleteReferencedGitRepo(Context.requireWorkspace().getId(), id);
+  }
+
+  @Override
+  protected void deleteControlled() {
+    logger.warn("Controlled git repo resource is not supported in workspace.");
```

Please throw an `UnsupportedOperationException` here instead.